SamplesDashboard/SamplesDashboard/Services/SampleService.cs

```diff
@@ -96,6 +96,8 @@ public async Task<List<Node>> GetSamples()
         private async Task SetHeaders(Node sampleItem) 
         {
             var headerDetails = await GetHeaderDetails(sampleItem.Name);
+            var repository = await GetRepository(sampleItem.Name);
```

Comment:
Since this function is not only setting headers, lets give it a more meaningful name now